source/PyMaterialX/PyMaterialXGenShader/PyShaderGenerator.cpp

```diff
@@ -190,16 +315,46 @@ class PyShaderGenerator : public mx::ShaderGenerator
             enumerationType
         );
     }
+
+  protected:
+    mx::ShaderStagePtr createStage(const std::string& name, mx::Shader& shader) const override
+    {
+        PYBIND11_OVERLOAD(
+            mx::ShaderStagePtr,
+            mx::ShaderGenerator,
+            createStage,
+            name,
+            shader
+        );
+    }
+
+    mx::ShaderNodeImplPtr createSourceCodeImplementation(mx::ImplementationPtr impl) const override
+    {
+        PYBIND11_OVERLOAD(
+            mx::ShaderNodeImplPtr,
+            mx::ShaderGenerator,
+            createSourceCodeImplementation,
+            impl
+        );
+    }
+
+    mx::ShaderNodeImplPtr createCompoundImplementation(mx::NodeGraphPtr impl) const override
+    {
+        PYBIND11_OVERLOAD(
+            mx::ShaderNodeImplPtr,
+            mx::ShaderGenerator,
+            createCompoundImplementation,
+            impl
+        );
+    }
 };
 
 void bindPyShaderGenerator(py::module& mod)
 {
     py::class_<mx::ShaderGenerator, PyShaderGenerator, mx::ShaderGeneratorPtr>(mod, "ShaderGenerator")
-        .def(py::init([](mx::SyntaxPtr syntax) { return new PyShaderGenerator(syntax); }))
```

Comment:
Are the methods listed here the ones being published? Just curious if we really need to declare all the other methods on the trampoline PyShaderGenerator class above then? This list looks good, and all the other methods above can maybe be removed?

Reply:
I think this says that they are all needed: https://pybind11.readthedocs.io/en/stable/advanced/classes.html#combining-virtual-functions-and-inheritance